src/DotNetWorker.Grpc/Handlers/InvocationHandler.cs

```
@@ -0,0 +1,154 @@
+                    {
+                        Name = binding.Key
+                    };
+
+                    if (binding.Value is not null)
+                    {
+                        parameterBinding.Data = await binding.Value.ToRpcAsync(_serializer);
+                    }
+
+                    response.OutputData.Add(parameterBinding);
+                }
+
+                if (functionBindings.InvocationResult != null)
+                {
+                    TypedData? returnVal = await functionBindings.InvocationResult.ToRpcAsync(_serializer);
+                    response.ReturnValue = returnVal;
+                }
+
+                response.Result.Status = StatusResult.Types.Status.Success;
+            }
+            catch (Exception ex)
+            {
+                response.Result.Exception = ex.ToRpcException();
+                response.Result.Status = StatusResult.Types.Status.Failure;
+
+                if (ex.InnerException is TaskCanceledException)
+                {
+                    response.Result.Status = StatusResult.Types.Status.Cancelled;
+                }
+            }
+            finally
+            {
+                _inflightInvocations.TryRemove(request.InvocationId, out var cts);
+
+                if (context is IAsyncDisposable asyncContext)
+                {
+                    await asyncContext.DisposeAsync();
+                }
+
+                (context as IDisposable)?.Dispose();
+            }
+
+            return response;
+        }
+
+        public void Cancel(string invocationId)
+        {
+            _inflightInvocations.TryGetValue(invocationId, out var cancellationTokenSource);
+            if (cancellationTokenSource is not CancellationTokenSource)
+            {
+                return;
+            }
+
+            try
+            {
+                cancellationTokenSource?.Cancel();
+            }
+            catch (Exception ex)
```

To reduce log noise, we can add a catch for `ObjectDisposedException` explicitly and skip logging there. Given the asynchronous nature of the `InvokeAsync` and cancellation messages, this will be an exception we encounter and should swallow as part of normal behavior.